polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,293 @@
+        """Delayed Geolocation Data write."""
+        msg = ("Adding geolocation 'longitude' and " "'latitude' datasets for grid %s", parent)
+        LOG.info(msg)
+        lon_data, lat_data = area_def.get_lonlats(chunks=chunks)
+
+        dtype = lon_data.dtype if dtype is None else dtype
+
+        lon_grp = "{}/longitude".format(parent)
+        lat_grp = "{}/latitude".format(parent)
+
+        lon_dataset = FakeHDF5(fname, lon_grp)
+        lon_dset = self.check_variable(fh, lon_grp, lon_data, dtype, append, compression)
+        lat_dataset = FakeHDF5(fname, lat_grp)
+        lat_dset = self.check_variable(fh, lat_grp, lat_data, dtype, append, compression)
+
+        return [lon_data, lat_data], [lon_dataset, lat_dataset]
+
+    @staticmethod
+    def create_variable(filename: str, parent, hdf_subgroup: str, dataset_id: xr.DataArray, dtype: np.dtype, **kwargs):
+        """Create a hdf5 subgroup and it's attributes."""
+
+        ds_attrs = dataset_id.attrs
+
+        if hdf_subgroup in parent:
+            LOG.warning("Product %s already in hdf5 group," "will delete existing dataset", hdf_subgroup)
+            del parent[hdf_subgroup]
+
+        try:
+            dset = parent.create_dataset(hdf_subgroup, shape=dataset_id.shape, dtype=dtype, **kwargs)
+        except ValueError:
+            if os.path.isfile(filename):
+                os.remove(filename)
+            raise
+
+        dset.attrs["satellite"] = ds_attrs["platform_name"]
+        dset.attrs["instrument"] = ds_attrs["sensor"]
+        dset.attrs["begin_time"] = ds_attrs["start_time"].isoformat()
+        dset.attrs["end_time"] = ds_attrs["end_time"].isoformat()
+
+        return
+
+    def save_datasets(
+        self,
+        dataset: list[xr.DataArray],
+        filename=None,
+        dtype=None,
+        fill_value=None,
+        append=True,
+        compute=True,
+        **kwargs,
+    ):
+        """Save hdf5 datasets."""
+
+        compression = kwargs.pop("compression", None) if "compression" in kwargs else None
+        if compression == "none":
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+
+        for dataset_id in dataset:
+            args = self._output_file_kwargs(dataset_id, dtype)
+            out_filename = filename or self.get_filename(**args)
+            output_names.append(out_filename)
+        one_file = all_equal(output_names)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
+            LOG.warning("Writing to {}".format(filename))
+
+        hdf5_fh = self.open_hdf5_filehandle(filename, append=append)
+
+        datasets_by_area = self.iter_by_area(dataset)
+        for area, data_arrs in datasets_by_area:
+            # Initialize source/targets at start of each new AREA grouping.
+            dsets = []
+            targets = []
```

Comment:
I think this is still wrong. If there is more than one area group then `targets` and `dsets` is overwritten when the second group is being processed and `return (dsets, targets)` at the bottom of this method will never include the earlier group(s).